### PR TITLE
Fetch playback URL for imported asset using source ID

### PR DESCRIPTION
**Pull request by yondonfu**
**What does this pull request do? Explain your changes. (required)**

Supersedes https://github.com/livepeer/studio/pull/1364 in favor of building off of https://github.com/livepeer/studio/pull/1365 which introduces a `source` field to the `Asset` entity.

This PR allows a user to specify a `source` field in the request body for a `/api/asset/import` request that looks like:

```
{
    "url": ...,
    "type": ...,
    "id": ...
}
```

For now, `type` must be `url`.

The user is responsible for specifying a value for the `id` field. For example, the value could be an IPFS CID or Arweave tx ID.

The `/playback` endpoint can then do a lookup for assets based on `source.id` which is useful when the asset is imported from an IPFS gateway and you want to plug in a CID to playback the asset. Or when the asset is imported from an Arweave gateway and you want to plug in a tx ID to playback the asset.

**Specific updates (required)**

See commit history.

- **How did you test each of these updates (required)**

Updated unit tests.

**Does this pull request close any open issues?**

Fixes https://github.com/livepeer/studio/issues/1362.

TODO: Test in an e2e environment and make sure /playback returns the correct playback URL for a source ID.

**Screenshots (optional):**

**Checklist:**

- [x] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: packages/api/src/store/asset-table.ts [hunk 1]
```diff
@@ -53,4 +53,19 @@ export default class AssetTable extends Table<WithID<Asset>> {
...
+      sql`asset.data->>'deleted' IS NULL`,
+    ];
+    const [assets] = await this.find(query, {
+      limit: 2,
+      order: "coalesce((asset.data->'createdAt')::bigint, 0) ASC",
+    });
+    if (!assets || assets.length < 1) {
+      return null;
+    }
+    return assets[0];
```
Comment (0xcadams) [inline packages/api/src/store/asset-table.ts:69]: Just calling out one unintended side-effect that may happen here - if two users create an asset with the same source CID, the first asset will have all views registered under it. I don't think this is a blocking issue, but something we'll need to iterate on.

File: packages/api/src/controllers/asset.ts [hunk 2]
```diff
@@ -485,11 +485,17 @@ const uploadWithUrlHandler: RequestHandler = async (req, res) => {
...
     Date.now(),
     req.config.vodObjectStoreId,
-    req.body
+    req.body,
+    req.body.source
   );
-  if (!req.body.url) {
+
+  const url =
+    req.body.url ?? (req.body.source ? req.body.source.url : undefined);
```
Comment (0xcadams) [inline packages/api/src/controllers/asset.ts:493]:
```suggestion
    req?.body?.url ?? req?.body?.source?.url;
```

**Review by 0xcadams — APPROVED** [review]

File: packages/api/src/store/asset-table.ts (hunk 1 above)
Comment (yondonfu) [reply packages/api/src/store/asset-table.ts:69]: Yeah good call out!

**Comment by yondonfu** [comment]
After a conversation in Discord with @victorges, I'm closing this PR in favor of https://github.com/livepeer/studio/pull/1370 which should be simpler approach and also ensures that the CIDs used to lookup playback URLs of imported IPFS assets actually match the asset data imported. Hope this is the last PR that I close for this issue :)

Cherry picked https://github.com/livepeer/studio/pull/1367/commits/7f0c972b2199d15b27a9b33490cede349ca22413 into https://github.com/livepeer/studio/pull/1370 as well.

**Pull request closed without merging**